agent/lib/kontena/workers/container_health_check_worker.rb

```diff
@@ -53,10 +55,16 @@ def handle_action(msg)
       if msg['status'] == 'unhealthy'
         name = @container.labels['io.kontena.container.name']
         # Restart the container, master will handle re-scheduling logic
-        info "About to restart container #{name} as it's reported to be unhealthy"
-        emit_service_pod_event("service:health_check", "restarting #{name} because it's reported as unhealthy", Logger::WARN)
-        
-        restart_container
+        info "#{name} is unhealthy"
+        emit_service_pod_event("service:health_check", "#{name} is unhealthy", Logger::WARN)
+
+        publish('container:health_check', Event.new(
```

Comment:
"Broadcasting" per-container health statuses feels bit of a wasted effort. These events should always be handled by a single pod worker. I could think of two possible alternatives:
- use something like `container:health_check:<service_id>:<instance_number>` as the channel key
- "route" the events through pod manager, it knows which pod worker should handle each event

Reply:
Or have the `ServicePodWorker` supervise the `ContainerHealthCheckWorkers` instead of spawning them from Docker events.

Then it can pass in `Celluloid.current_actor`, and the health check worker can send these as direct calls to the pod worker.

Reply:
yep, that's even better I think. that way these are also tied to each others lifecycle nicely.